### ISSUE TITLE
Own Colshape factory broken

### ISSUE THREAD

**Issue by TheGodlyLuzer** [altmp/altv-issues#1473]
### Description of the problem

_*Not sure if this is my error or a bug, but i thought i might report it since i can't find a mistake on my side._

When using my own ColShape class "DataColshape", the ColshapeEvent returns `DataColshape` but all the values are set to default I tried logging the  ```.ColShapeNativePointer``` when the Colshape is created and when a collision occurs, they are the same.

If you log the stored value right after creating the Colshape, everything returns correctly. Even setting an interval to log it every x seconds returns the correct data. But not on the colision

### Reproduction steps

Create ColShape Factory
```cs
public class DataColshapeFactory : IBaseObjectFactory<IColShape>
{
    public IColShape Create(ICore server, IntPtr colshapePointer)
    {
        return new DataColshape(server, colshapePointer);
    }
}
```

Create `DataColshape` Class
```cs
public class DataColshape : ColShape
{
	public uint SpotId { get; set; }
	public ColshapeType.Types ColshapeType { get; set; }
	
	public DataColshape(ICore server, IntPtr nativePointer) : base(server, nativePointer)
	{
	}
}
```

Create new Colshape
```cs
var shape = (DataColshape)Alt.CreateColShapeCircle(new Position(spot.X, spot.Y, 0), spot.Radius);
shape.SpotId = spot.Id;
shape.ColshapeType = ColshapeType.Types.TestSpot;
```

`OnColshapeEvent` log `colshape.SpotId` -> always returns 0.

### Expected behaviour

When entering the colshape, the `DataColshape` should return the stored values, not the default values.

### Operating system

Windows 10

### Version

10.0-rc1

**Comment by deluvas1911** [comment]
Related to C# module

**Comment by zziger** [comment]
Fixed in C# module 10.0.2-rc
